save-backend/src/main/kotlin/com/saveourtool/save/backend/security/ProjectPermissionEvaluator.kt

```diff
@@ -82,16 +100,19 @@ class ProjectPermissionEvaluator(
             Mono.error(ResponseStatusException(HttpStatus.NOT_FOUND))
         }
 
-    private fun hasReadAccess(userId: Long?, projectRole: Role): Boolean = hasWriteAccess(userId, projectRole) ||
+    private fun hasReadAccess(userId: Long?, projectRole: Role, organzationRole: Role): Boolean = hasWriteAccess(userId, projectRole, organzationRole) ||
             userId?.let { projectRole == Role.VIEWER } ?: false
 
-    private fun hasWriteAccess(userId: Long?, projectRole: Role): Boolean = hasDeleteAccess(userId, projectRole) ||
+    private fun hasWriteAccess(userId: Long?, projectRole: Role, organzationRole: Role): Boolean = hasDeleteAccess(userId, projectRole, organzationRole) ||
             userId?.let { projectRole == Role.ADMIN } ?: false
 
-    private fun hasDeleteAccess(userId: Long?, projectRole: Role): Boolean =
-            hasBanAccess(userId, projectRole) || userId?.let { projectRole == Role.OWNER } ?: false
+    private fun hasDeleteAccess(userId: Long?, projectRole: Role, organzationRole: Role): Boolean =
+            hasBanAccess(userId, projectRole, organzationRole) ||
+                    userId?.let { projectRole == Role.OWNER || organzationRole == Role.OWNER } ?: false
 
-    private fun hasBanAccess(userId: Long?, projectRole: Role): Boolean = userId?.let { projectRole == Role.SUPER_ADMIN } ?: false
+    private fun hasBanAccess(userId: Long?, projectRole: Role, organzationRole: Role): Boolean = userId?.let {
+        projectRole == Role.SUPER_ADMIN || organzationRole == Role.SUPER_ADMIN
```

Comment:
It is never so btw as far as `Role.SUPER_ADMIN` cannot be assigned on organization/project level. It is a special role that is defined on a higher level.

Reply:
Done